src/test/java/http/router/RouterTest.java

```diff
@@ -0,0 +1,89 @@
+package http.router;
+
+import http.handlers.request.*;
+import http.request.HttpRequest;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class RouterTest {
+
+    private HttpRequest httpRequestGet,
+                        httpRequestPost,
+                        httpRequestDelete,
+                        httpRequestOptions,
+                        httpRequestPut,
+                        httpRequestInvalid;
+
+    @BeforeEach
+    void setUp() {
+        httpRequestGet = new HttpRequest(rawGetRequest());
+        httpRequestPut = new HttpRequest(rawPutRequest());
+        httpRequestPost = new HttpRequest(rawPostRequest());
+        httpRequestDelete = new HttpRequest(rawDeleteRequest());
+        httpRequestOptions = new HttpRequest(rawOptionsRequest());
+        httpRequestInvalid = new HttpRequest(rawInvalidRequest());
+    }
+
+
+
+    @Test
+    void getResponse() {
+        assertEquals(DirectoryHandler.class, Router.getHandler(httpRequestGet).getClass());
+        assertEquals(PutRequestHandler.class, Router.getHandler(httpRequestPut).getClass());
+        assertEquals(PostRequestHandler.class, Router.getHandler(httpRequestPost).getClass());
+        assertEquals(DeleteRequestHandler.class, Router.getHandler(httpRequestDelete).getClass());
+        assertEquals(OptionsRequestHandler.class, Router.getHandler(httpRequestOptions).getClass());
+        assertEquals(InvalidRequestHandler.class, Router.getHandler(httpRequestInvalid).getClass());
+    }
+
+
+    private String rawGetRequest() {
+        return "GET / HTTP/1.1\r\n" +
+                "Host: www.nowhere123.com\r\n" +
```

Comment:
All these headers look the same in each helper function here. Can we move them into a String? So we have something like return "GET / HTTP/1.1\r\n" + headers;

Also, are those headers needed to pass the test?

Reply:
@td-extreme Those headers are not needed to pass the test. I'll remove them.